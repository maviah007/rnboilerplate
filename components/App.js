// @flow

import React, { Component } from 'react';
import { StyleSheet, View } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
});

type Props = {
  receiver: string
};

type State = {
  value: number
};

export default class App extends Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      value: 0,
    };
  }
  render() {
    return (
      <View style={styles.container}>
        {this.props.receiver}
        {this.state.value}
      </View>
    );
  }
}
