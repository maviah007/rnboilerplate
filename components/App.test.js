import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('Testing App component', () => {
  it('renders as expected', () => {
    const wrapper = shallow(<App receiver="Hello" />);
    expect(wrapper).toMatchSnapshot();
    wrapper.setProps({ receiver: 'Bye' });
    expect(wrapper).toMatchSnapshot();
  });
});
