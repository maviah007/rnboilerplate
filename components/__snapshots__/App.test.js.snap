// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App component renders as expected 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  Hello
  0
</Component>
`;

exports[`Testing App component renders as expected 2`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  Bye
  0
</Component>
`;
